src/ascii_str.rs

```diff
@@ -1079,7 +1167,8 @@ mod tests {
         assert_eq!(generic(ascii_str), Ok(ascii_str));
         assert_eq!(generic(&"A"), Ok(ascii_str));
         assert_eq!(generic(&ascii_str), Ok(ascii_str));
-        assert_eq!(generic(&mut "A"), Ok(ascii_str));
```

Comment:
I guess this tests whether `AsAsciiStr` is implemented by all expected types. Therefore it should be:
```suggestion
        // We need a second copy to prevent overlapping mutable borrows.
        let mut mut_arr = arr;
        let mut mut_ascii_str: &mut AsciiStr = mut_arr.as_mut().into();
        assert_eq!(generic(&&mut "A"), Ok(ascii_str));
```
When written like that it detects a missing `impl AsAsciiStr for &'a mut AsciiStr` impl.

Reply:
Oh, that makes sense. So we'd test for both `impl AsAsciiStr &'a &mut {str, AsciiStr}`.
I'll go implement that test now for `str`, `AsciiStr` and I suppose `[AsciiChar]` too, since it implements `ToAsciiStr`.

Reply:
I think I fixed the test, I suppose now that I made a change I have to rebase on master because of the conflicts?

Reply:
We could also use the trick I saw in the `static-assertions` crate:
```rust
    fn assert_impl_all<T: ?Sized + AsAsciiStr>() {}
    assert_impl_all::<&str>();
    assert_impl_all::<&mut str>();
    ...
```
This works at compile time.

Reply:
Mh, that should not be necessary. Are there any conflicts? Since you already rebased this branch on master it wouldn't change anything. You should be able to just commit the new change.

Reply:
I'm still relatively new to git, so I assumed to merge I had to rebase to the latest master? At least down at the bottom of the github PR it says there are conflicting changes with `src/ascii_string.rs`

Reply:
Ah, you're right. There are conflicts due to the other PR I merged. Yes, then you can solve the conflict by rebasing and editing the conflict. When the rebase prompts you to edit your conflict, edit it, `git add` the resolved file and `git rebase --continue`.

You could alternatively merge master into your branch, which would prompt you also to resolve conflicts. Since this creates new edges between branches in the commit graph I prefer the rebase option.

Reply:
I think I've rebased it, the only issue was the `AsciiString::new` being const fn now, but that also brought another clippy lint, about the method now being a candidate for `#[must_use]`, so I'll add that in a new commit.